src/contrib/cluster/Akka.Cluster.Metrics/Routing/ClusterMetricsRouting.cs

```diff
@@ -328,7 +333,7 @@ public override Router CreateRouter(ActorSystem system)
         /// <inheritdoc />
         public override Props RoutingLogicController(RoutingLogic routingLogic)
         {
-            return Actor.Props.Create(() => new AdaptiveLoadBalancingMetricsListener(routingLogic as AdaptiveLoadBalancingRoutingLogic));
+            return Actor.Props.Create(() => new AdaptiveLoadBalancingMetricsListener((IClusterMetricsRoutingLogic)routingLogic));
```

Comment:
Blowing up here is definitely better than doing it as an NRE inside the `AdaptiveLoadBalancingMetricsListener`, IMHO.